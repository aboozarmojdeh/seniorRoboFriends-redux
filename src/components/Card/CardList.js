import React, { Fragment } from 'react';
import Card from './Card';
const CardList = ({ robots }) => {
    // if(true){
    //     throw new Error('NOOOOOOOOOOOO')
    // }
    const cardArray = robots.map((robot, i) => {
        return (
            <Card key={robot.id} id={robot.id} name={robot.name} email={robot.email} link={robot.link} />
        )
    })
    return (
        <Fragment>
        <div>
            {cardArray}
        </div>
        </Fragment>
    )
};

export default CardList;