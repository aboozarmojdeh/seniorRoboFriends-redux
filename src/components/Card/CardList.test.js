import React from 'react';
import { shallow, mount, render } from 'enzyme';
// import renderer from 'react-test-renderer';
import CardList from './CardList';

// npm test -- --coverage --watchAll=false



const mockRobots = [
    {
      id: 1,
      name: 'Leanne Graham',
      username: 'Bret',
      email: 'Sincere@april.biz',
      link:'https://www.google.com'
    },
    {
      id: 2,
      name: 'Ervin Howell',
      username: 'Antonette',
      email: 'Shanna@melissa.tv',
      link:'https://www.google.com'
    },
    {
      id: 3,
      name: 'Clementine Bauch',
      username: 'Samantha',
      email: 'Nathan@yesenia.net',
      link:'https://www.google.com'
    }
  ];

  const wrapper = shallow(<CardList robots={mockRobots} />); 
  
test('expect to render CardList component',()=>{
    expect(wrapper).toMatchSnapshot();
})


