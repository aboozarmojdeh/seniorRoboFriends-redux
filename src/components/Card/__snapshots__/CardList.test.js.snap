// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
<Fragment>
  <div>
    <Card
      email="Sincere@april.biz"
      id={1}
      key="1"
      link="https://www.google.com"
      name="Leanne Graham"
    />
    <Card
      email="Shanna@melissa.tv"
      id={2}
      key="2"
      link="https://www.google.com"
      name="Ervin Howell"
    />
    <Card
      email="Nathan@yesenia.net"
      id={3}
      key="3"
      link="https://www.google.com"
      name="Clementine Bauch"
    />
  </div>
</Fragment>
`;
